woocommerce/woocommerce-gutenberg-products-block
# Add product query support for Sale badge block

Adds Product Query support for the atomic Sale badge block

On the client side, when the Sale badge block is used within the product query block, the markup will be rendered on the server side - No javascript related to Sale badge block will be rendered.

Fixes #7331 

#### Testing
1. Checkout this branch and run npm start
2. Be sure that you have latest version of the Gutenberg feature plugin installed and activated
3. Create a post/page and add the Product Query block.
4. Add `Sale Badge` block to Post Template of product query block as shown in video below:
https://user-images.githubusercontent.com/16707866/202644003-ee85857b-a01f-4b4f-819a-ec910b630dd5.mov
5. Now save the post & Open the frontend side and confirm that: 
 - No JS related to the Product Sale Badge block is loaded (search in your network tab for product-sale-badge-frontend.js). As you can see in the screenshot below, `product-sale-badge-frontend.js` showing before the changes made in this PR. This shouldn't be visible in network tab anymore.
<img width="865" alt="image" src="https://user-images.githubusercontent.com/16707866/202644926-7f4be5d4-1bab-4b3d-ae9f-80586df8df98.png"> 

 - The UI of the block is identical to on the editor side.
 - That the settings are reflected correctly (eg: custom style). Try changing custom style from the sidebar(Shown in screenshot below): 
<img width="289" alt="image" src="https://user-images.githubusercontent.com/16707866/202645737-954c18e1-32ad-4096-a080-007e6e0d446f.png">

 - `ADDITIONAL CSS CLASS(ES)`(available in advanced toggle in sidebar) should be added to the container div
<img width="273" alt="image" src="https://user-images.githubusercontent.com/16707866/202675897-6478bedf-d02c-44d5-b9e1-105582c5e1c3.png">


### WooCommerce Visibility
